### PR TITLE
Feat: Extend MapExplorer to show CPM

### LINKED ISSUE #1007: Statewise heat-map based on Population density
**Is your feature request related to a problem? Please describe.**

State with a large population will have more cases compared to one with less population. I have seen multiple websites (Other countries) using the number of infected persons per million population criteria to plot heatmap. 

This will help in visualizing district-wide hotspots. 

If You compare UP, Maharastra and Kerala. the ratio of infected Population/total Population

- UP: 0.22 cases per million population
- Kerala: 1.12 cases per million population
- Delhi: 6.38 cases per million population
- Maharastra: 1.69 cases per million population

So, Delhi is more affected than Maharashtra but In count. Maharastra is most affected. 

We can go granular to find more hotspots (District or city level). This will help in understanding the affected cities in a more logical way.

### PR DESCRIPTION
**Pull request by nishcthulhu**
**Description of PR**
#1007 Statewise heat-map based on population density (cases per million)
**Type of PR**
- [x] New feature

**Relevant Issues**  
Fixes #1007 

**Checklist**

- [x] Compiles and passes lint tests
- [x] Properly formatted
- [x] Tested on desktop
- [x] Tested on phone

**Screenshots**
Screenshot 2020-04-18 at 5 07 33 PM

### CODE REVIEW

**Comment by jeremyphilemon** [comment]
Hey can you check this again, looks like the state pages aren't working? I really like the implementation btw.

Also any ideas on how to show the number for each of the states too, the map does a good job at giving a nice visual summary, but showing numbers all would be great! Thoughts?

**Comment by nishcthulhu** [comment]
Hey I'm trying to figure out how to include the case per million value to display on the top left of the map

But could you tell me how to reproduce the state page breaking error? I'm unable to reproduce it for some reason

Data for population is from census India dashboard. Have split it manually for ladakh and telangana at their constituent district level as they were formed post the census

**Comment by nishcthulhu** [comment]
@jeremyphilemon If by the state page u mean showing case denjsty by district then the challenge there is getting the data.

So I have decided not to have that option for state pages

**Comment by nishcthulhu** [comment]
Have made the case density per million button invisible considering that options is not available as of now

**Comment by nishcthulhu** [comment]
Addedcallback function on state page as well to prevent breaking

File: src/components/choropleth.js [hunk 1]
```diff
@@ -72,13 +73,21 @@ function ChoroplethMap({
       const margin = {left: 0.02 * widthLegend, right: 0.02 * widthLegend};
       const barWidth = widthLegend - margin.left - margin.right;
       const heightLegend = +svgLegend.attr('height');
+      let title = 'Confirmed Cases';
+
+      if (type === 1) {
+        title = 'Confirmed Cases';
+      } else {
+        title = 'Case Density Per Million';
+      }
```
Comment (Maskedman99) [inline src/components/choropleth.js]: ```title = type === 1 ? 'Confirmed Cases' : 'Case Density Per Million' ;```

Comment (nishcthulhu) [reply src/components/choropleth.js]: Future proofs it in case we add multiple views on future such as "tests per million"

**Comment by nishcthulhu** [comment]
@jeremyphilemon can you please help get this merged

**Comment by nishcthulhu** [comment]
@JunaidBabu @sudevschiz can any of you help.out in checking is this good to merge?

**Comment by bee-rickey** [comment]
@nishcthulhu Did you add anything to show the actual value anywhere? Since the actual ppm is not visible, it's not clear what the value is.

**Comment by nishcthulhu** [comment]
Updated on hover it now shows the cases per million in each state 

@jeremyphilemon @bee-rickey @JunaidBabu @sudevschiz please merge if approved

Screenshot 2020-04-21 at 1 44 18 PM

**Comment by nishcthulhu** [comment]
@jeremyphilemon the branch is now up to date with the latest master and  the new refactored code base.

Please merge to enable this deep  dive view

**Comment by nishcthulhu** [comment]
@jeremyphilemon any idea when this PR can be merged? I'm seeing a few new features being implemented on the MapExplorer and was hoping  if this gets merged those can be added on, else this code would need to be refactored all over again (e.g: https://github.com/covid19india/covid19india-react/pull/1407)

**Comment by jeremyphilemon** [comment]
Hey @nishcthulhu, sorry about the delay and I thank your patience. We're still reviewing this PR, everything else looks fine on the code side btw! I'll update you here as we progress :)

Please bear with us a little bit longer!

**Comment by shuklaayush** [comment]
@nishcthulhu Gujarat is not the state with the highest CPM.  Can you cross-check the population data? Seems to be off by an order of magnitude for some states.

**Comment by nishcthulhu** [comment]
I pulled this data from the official 2011 census database as that's the only trustworthy source

By those numbers Gujarat does have the highest cpm considering they have over half the cases of Maharashtra with a much lower population base

**Comment by nishcthulhu** [comment]
@shuklaayush u can see the data here 
http://censusindia.gov.in/2011-common/census_info.html

**Comment by shuklaayush** [comment]
I'm telling you it's wrong. Gujarat's population ~ 6 crores. What you have in the file is 6383628 (~0.63 crores)

**Comment by nishcthulhu** [comment]
> I'm telling you it's wrong. Gujarat's population ~ 6 crores. What you have in the file is 6383628 (~0.63 crores)

@shuklaayush yup there was an error in parsing the data.

It's updated now

**Comment by shuklaayush** [comment]
Hey @nishcthulhu, can you rebase this on latest master and fix the percentages shown on the tooltip?

**Comment by nishcthulhu** [comment]
@shuklaayush its been rebased

**Comment by nishcthulhu** [comment]
@jeremyphilemon @shuklaayush could you please deploy this if it looks good to you guys?

**Comment by nishcthulhu** [comment]
Hey @shuklaayush, I keep on committing changes to this PR and it is not getting merged. Post that other PRs get committed and I need to rebase it again to remove conflicts.

Could you please tell me if this feature has been deprecated otherwise this leads to rework with no feedback

**Comment by shuklaayush** [comment]
@nishcthulhu Just merged this. Thanks for your contribution! :)

**Pull request merged**